jbmc/src/java_bytecode/generic_parameter_specialization_map_keys.cpp

```diff
@@ -44,43 +58,34 @@ void generic_parameter_specialization_map_keyst::insert_pairs(
   const std::vector<reference_typet> &types)
 {
   INVARIANT(erase_keys.empty(), "insert_pairs should only be called once");
-  PRECONDITION(parameters.size() == types.size());
 
-  // Pair up the parameters and types for easier manipulation later
-  std::vector<std::pair<java_generic_parametert, reference_typet>> pairs;
-  pairs.reserve(parameters.size());
-  std::transform(
-    parameters.begin(),
-    parameters.end(),
-    types.begin(),
-    std::back_inserter(pairs),
-    [&](java_generic_parametert param, reference_typet type)
-    {
-      return std::make_pair(param, type);
-    });
-
-  for(const auto &pair : pairs)
+  for(const auto &pair : zip(parameters, types))
   {
     // Only add the pair if the type is not the parameter itself, e.g.,
     // pair.first = pair.second = java::A::T. This can happen for example
-    // when initiating a pointer to an implicitly java generic class type
+    // when initializing a pointer to an implicitly generic java class type
     // in gen_nondet_init and would result in a loop when the map is used
     // to look up the type of the parameter.
     if(
-      !(is_java_generic_parameter(pair.second) &&
-        to_java_generic_parameter(pair.second).get_name() ==
-          pair.first.get_name()))
+      is_java_generic_parameter(pair.second) &&
```

Comment:
Personally I'd say `if(condition) thing` is clearer than `if(!condition) continue; thing`, even if the condition is negated. If you think the negative is the source of inclarity perhaps just name the condition (`bool name_matches = ...; if (!name_matches) thing` ?)

Reply:
I disagree, but will revert to avoid creating churm for something that is apparently a matter of opinion.